### PR TITLE
[Lock] add aliases for LockFactory

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       |4.4
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | none   
| License       | MIT
| Doc PR        | none 

see https://github.com/symfony/symfony/pull/32495/files#r302844198, add missing aliases in 4.4 and in 5.0 we just remove the old classes and aliases.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/lock.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
         <service id="lock.strategy.majority" class="Symfony\Component\Lock\Strategy\ConsensusStrategy" />
 
-        <service id="lock.factory.abstract" class="Symfony\Component\Lock\Factory" abstract="true">
+        <service id="lock.factory.abstract" class="Symfony\Component\Lock\LockFactory" abstract="true">
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/lock.xml]: we also need to conflict with lock < 4.4 and bump in require-dev

Comment (Simperfit) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/lock.xml]: done

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 2]
```diff
@@ -81,7 +81,8 @@
         "symfony/translation": "<4.3",
         "symfony/twig-bridge": "<4.1.1",
         "symfony/validator": "<4.1",
-        "symfony/workflow": "<4.3"
+        "symfony/workflow": "<4.3",
+        "symfony/lock": "<4.4"
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: alpha order

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 3]
```diff
@@ -57,7 +57,7 @@
         "symfony/workflow": "^4.3|^5.0",
         "symfony/yaml": "^3.4|^4.0|^5.0",
         "symfony/property-info": "^3.4|^4.0|^5.0",
-        "symfony/lock": "^3.4|^4.0|^5.0",
+        "symfony/lock": "^4.4|^5.0",
```
Comment (jderusse) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: Why?

Comment (jderusse) [reply src/Symfony/Bundle/FrameworkBundle/composer.json]: I mean, couldn't we inject the right class in the extension?

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/composer.json]: Because KISS :)

**Review by xabbuh — APPROVED** [review]

**Review by jderusse — APPROVED** [review]

**Review by oliverquynh — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @Simperfit.

**Pull request merged**